### ISSUE TITLE
Crash when throwing an error in app.on('web-contents-created')

### ISSUE THREAD

**Issue by nornagon** [electron/electron#28749]
Repro gist: https://gist.github.com/2b0373a300af0931c0c29b0e8dff8b73

Stack trace:

```
[39005:0420/121556.035182:FATAL:electron_api_browser_window.cc(71)] Check failed: web_contents_view.get(). 
0   Electron Framework                  0x000000010ba38aa9 base::debug::CollectStackTrace(void**, unsigned long) + 9
1   Electron Framework                  0x000000010b955563 base::debug::StackTrace::StackTrace() + 19
2   Electron Framework                  0x000000010b96e07f logging::LogMessage::~LogMessage() + 175
3   Electron Framework                  0x000000010b96ef4e logging::LogMessage::~LogMessage() + 14
4   Electron Framework                  0x0000000106cdffdd electron::api::BrowserWindow::BrowserWindow(gin::Arguments*, gin_helper::Dictionary const&) + 765
5   Electron Framework                  0x0000000106ce24c0 electron::api::BrowserWindow::New(gin_helper::ErrorThrower, gin::Arguments*) + 368
6   Electron Framework                  0x0000000106ce2b42 base::internal::Invoker<base::internal::BindState<gin_helper::WrappableBase* (*)(gin_helper::ErrorThrower, gin::Arguments*)>, gin_helper::WrappableBase* (gin_helper::ErrorThrower, gin::Arguments*)>::Run(base::internal::BindStateBase*, gin_helper::ErrorThrower&&, gin::Arguments*) + 34
7   Electron Framework                  0x0000000106ce2af8 gin_helper::WrappableBase* gin_helper::internal::InvokeFactory<gin_helper::ErrorThrower, gin::Arguments*>(gin::Arguments*, base::RepeatingCallback<gin_helper::WrappableBase* (gin_helper::ErrorThrower, gin::Arguments*)> const&) + 152
8   Electron Framework                  0x0000000106ce29b1 void gin_helper::internal::InvokeNew<gin_helper::WrappableBase* (gin_helper::ErrorThrower, gin::Arguments*)>(base::RepeatingCallback<gin_helper::WrappableBase* (gin_helper::ErrorThrower, gin::Arguments*)> const&, v8::Isolate*, gin_helper::Arguments*) + 81
9   Electron Framework                  0x0000000106cd0624 gin_helper::Dispatcher<void (v8::Isolate*, gin_helper::Arguments*)>::DispatchToCallback(v8::FunctionCallbackInfo<v8::Value> const&) + 356
10  Electron Framework                  0x00000001088fc1a2 v8::internal::FunctionCallbackArguments::Call(v8::internal::CallHandlerInfo) + 994
11  Electron Framework                  0x00000001088f9bd0 v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<true>(v8::internal::Isolate*, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::FunctionTemplateInfo>, v8::internal::Handle<v8::internal::Object>, v8::internal::BuiltinArguments) + 928
12  Electron Framework                  0x00000001088f8d1f v8::internal::Builtin_Impl_HandleApiCall(v8::internal::BuiltinArguments, v8::internal::Isolate*) + 383
13  Electron Framework                  0x00000001088f890d v8::internal::Builtin_HandleApiCall(int, unsigned long*, v8::internal::Isolate*) + 109
14  ???                                 0x0000004600318bdf 0x0 + 300650957791
```

it's a DCHECK, but I'm pretty sure it will crash later if this is null.

**Comment by codebytere** [comment]
Yeah if it's run on the latest nightly:
```
Received signal 11 SEGV_MAPERR 000000000058
 [0x00010c7c55b9]
 [0x00010fdbb6d3]
 [0x00010fdd702b]
 [0x7fff73d605fd]
 [0x7ffee4fa2660]
 [0x00010acfb4f1]
 [0x00010acfd5f8]
 [0x00010acfdd01]
 [0x00010acfdc1e]
 [0x00010acf07de]
 [0x00010b6457c0]
 [0x00010be56558]
 [0x00010bdeb541]
[end of stack trace]
```

**Comment by VerteDinde** [comment]
Traced this back to Electron 10.0.0-beta.1 (latest 9 loads with a console error, but no crash). This seems to be the breaking change: https://github.com/electron/electron/pull/22997

**Comment by MarshallOfSound** [comment]
@VerteDinde Yeah this appears to be a case of us calling V8 APIs with a pending exception on the isolate.  I think the correct way to fix this long term is to to wrap all usages of `CallMethodWithArgs` in `event_emitter_caller.h` with a `v8::TryCatch` and send the error to the global error handler ourselves manually 🤔  Effectively declare it an uncaught exception.

The exact case that causes a crash is this:

* User JS code triggers some native code
* That native code emits an event
* That event is emitted in the same scope as the JS code that triggered the native code
* The event handler throws an error
* The native code then tries to run more JS (emit another event, call a getter, etc.) which is against the rules because there is a pending exception

This doesn't happen for most events because most events Electron emits are not triggered by JS code they are instead the start of their own stack and an error inside a handler will instantly go to the unhandled exception handler.  This case is unique because it's in the constructor of a gin wrappable 🤔 

TLDR: Let's put some try catches in here 😄
